src/main/java/org/threadly/examples/features/PrioritySchedulerExample.java

```diff
@@ -0,0 +1,146 @@
+package org.threadly.examples.features;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.threadly.concurrent.future.FutureUtils;
+import org.threadly.concurrent.future.ListenableFuture;
+import org.threadly.concurrent.PriorityScheduler;
+import org.threadly.concurrent.TaskPriority;
+
+
+/**
+ * Provides a basic example of the PriorityScheduler feature.
+ * 
+ * @author sreid8 - Sean Reid
+ *
+ */
+public final class PrioritySchedulerExample {
+  
+  /** static reference to the futures created */ 
+  private static List<ListenableFuture<?>> statFutures;
+  
+  /**
+   * Provided a PriorityScheduler that contains some basic ideas
+   * @return -> the PriorityScheduler with tasks added
+   */
+  public static PriorityScheduler getPriorityScheduler() {
+    final int numThreads = Runtime.getRuntime().availableProcessors() * 2;
+    final PriorityScheduler executor = new PriorityScheduler(numThreads);
+    List<ListenableFuture<?>> futures = new ArrayList<ListenableFuture<?>>(numThreads);
+    
+    futures.add(executor.submit(new Runnable() {
+      
+      @Override
+      public void run() {
+        /*
+         * This is the Starvable priority task that will be run.
+         * Starvable tasks run only when no other task needs to run
+         * It can be used to keep track of configuration:
+         * Example use case: closing sockets that no longer have hosts
+         * available on the other side.
+         */
+        
+      }
+    },
+    /*The priority of the task we are submitting*/TaskPriority.Starvable));
+    
+    futures.add(executor.submit(new Runnable() {
+      
+      @Override
+      public void run() {
+        /*
+         * This is the Low priority task that will be run.
+         * Low priority tasks are run before Starvable tasks,
+         * but exactly when it will be run is enforced by the 
+         * PrioritySchedulerService.
+         * Example use case: cache cleaning or other cleanup work
+         */
+        
+      }
+    }, TaskPriority.Low));
+    
+    futures.add(executor.submit(new Runnable() {
+
+      @Override
+      public void run() {
+        /*
+         * This is the High priority task that will be run.
+         * High priority tasks are the highest priority in the thread pool.
+         * Example use case: Stats reporting about number of threads active in the pool
```

Comment:
I think the best example of high priority tasks is servicing the actual requests of an application, or doing processing work where the result needs to be produced as soon as possible.